### PR TITLE
Fix PeerManager not to accept connections from already connected peers

### PR DESCRIPTION
**Pull request by LukasGasior1**

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala [hunk 1]
```diff
@@ -104,10 +107,15 @@ class PeerManagerActor(
...
+        context.unbecome()
+        val peer = createPeer(remoteAddress)
+        peer.ref ! PeerActor.HandleConnection(connection, remoteAddress)
+        unstashAll()
+      } else {
+        log.info("Another connection with {} is already opened. Disconnecting.", remoteAddress)
+        connection ! PoisonPill
+      }
 
     case (HandlePeerConnection(connection, remoteAddress), Failure(_)) =>
```
Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala]: As in this case we are also establishing a connection with a new peer, should we also check whether  `isConnectionHandled` is true or not?

**Comment by ntallar** [comment]
As when receving `HandlePeerConnection` and `ConnectToPeer` messages during `listening` we disconnect from peers if there are too many (when calling `tryDiscardPeersToFreeUpLimit()`), maybe it would be better to check there for `isConnectionHandled` is true or not? That way we wouldn't be disconnecting from other peers to make room to a duplicated peer we won't end connecting to.

**Comment by rtkaczyk** [comment]
@LukasGasior1 apart from Nico's comments, LGTM

File: src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala (hunk 1 above)
Comment (LukasGasior1) [inline src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala]: In this case the peer limit has been reached (note, the `Failure` in the tuple). The peer will be disconnected and closed anyway (see `       peer.ref ! PeerActor.DisconnectPeer(Disconnect.Reasons.TooManyPeers)` few lines below)

**Pull request merged**